lib/signaling/v2/room.js

```diff
@@ -659,13 +662,17 @@ function handleLocalParticipantEvents(roomV2, localParticipant) {
   localParticipant.on('trackRemoved', renegotiate);
   localParticipant.on('updated', localParticipantUpdated);
 
-  roomV2.on('stateChanged', function stateChanged(state) {
+  roomV2.on('stateChanged', function stateChanged(state, error) {
     if (state === 'disconnected') {
       localParticipant.removeListener('trackAdded', renegotiate);
       localParticipant.removeListener('trackRemoved', renegotiate);
       localParticipant.removeListener('updated', localParticipantUpdated);
       roomV2.removeListener('stateChanged', stateChanged);
       localParticipant.disconnect();
+    } else if (state === 'reconnecting' && error instanceof SignalingConnectionDisconnectedError) {
```

Comment:
is it possible to have `(state === 'reconnecting && !error)` - it would be useful to assert or log.error in such case.

Reply:
@makarandp0 We need to make sure LocalParticipantV2 transitions to "reconnecting" only if the `error` is an instance of `SignalingConnectionDisconnectedError` (signaling connection interruption).

Reply:
ah make sense. maybe then handling the "other" case with a comment would help.